tests/v2/test_stack_aggregator.py

```diff
@@ -319,3 +350,44 @@ def test_gremlin_batch_call(_mock_gremlin):
     assert (2, 3, 2) == _gremlin_batch_test(_mock_gremlin, 3)
     assert (2, 4, 1) == _gremlin_batch_test(_mock_gremlin, 4)
     assert (1, 0, 5) == _gremlin_batch_test(_mock_gremlin, 5)
+
+
+class TestStackAggregator(TestCase):
+    """Test for the Stack Aggregator class."""
+
+    def _mocked_get_data_from_db(*args):
+        if args[3] == 'pckg_response':
+            with open("tests/v2/data/golang_pkg_node_gremlin_response.json", "r") as fin:
+                return json.load(fin)
+        elif args[3] == 'module_vulnerabilities':
+            with open("tests/v2/data/golang_module_vuls_graph_response.json", "r") as fin:
+                return json.load(fin)
+        elif args[3] == '_get_pkg_details_with_vuls':
+            with open("tests/v2/data/graph_response_2_public_vuln.json", "r") as fin:
+                return json.load(fin)
+
+    @mock.patch('src.v2.stack_aggregator.GoAggregator._get_data_from_graph',
+                side_effect=_mocked_get_data_from_db, autospec=True)
+    @mock.patch('src.v2.stack_aggregator.get_license_analysis_for_stack')
+    def test_golang_pseudo_version(self, _mock_lic, _mock_get_data_from_db):
+        """Test Golang Pseudo Version functionality."""
+        resp = StackAggregator().execute(_go_request_body(), persist=False)
+        _mock_get_data_from_db.assert_called()
+        _mock_lic.assert_called_once()
+        self.assertEqual(resp['aggregation'], 'success')
+        self.assertEqual(len(resp['result']['unknown_dependencies']), 1)
+        self.assertEqual(len(resp['result']['analyzed_dependencies']), 1)
+        self.assertEqual(len(resp['result']['analyzed_dependencies'][0]
+                             ['dependencies']), 1)
+        self.assertEqual(len(resp['result']['analyzed_dependencies'][0]
+                             ['public_vulnerabilities']), 3)
+        list_of_keys = resp['result'].get(
+            'analyzed_dependencies')[0]['public_vulnerabilities'][0].keys()
+        self.assertIn('exploit', list_of_keys)
+        self.assertIn('fixable', list_of_keys)
+        self.assertIn('severity', list_of_keys)
+        self.assertIn('fixed_in', list_of_keys)
+        self.assertEqual(len(resp['result'].get(
+            'analyzed_dependencies')[0]['private_vulnerabilities']), 0)
+        self.assertIsInstance(resp['result']['registration_link'], str)
+        self.assertEqual(resp['result']['ecosystem'], 'golang')
```

Comment:
You should also assert the expected values, not just length.

Reply:
Ack